### ISSUE TITLE
Android Kill Switch Not Working

### ISSUE THREAD

**Issue by znaf37g** [AdguardTeam/AdGuardVPNForAndroid#508]
### Please answer the following questions for yourself before submitting an issue.

- [X] I checked the [knowledge base](https://adguard-vpn.com/kb/) and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard VPN version

2.10.145

### Operating mode

Integration with AdGuard

### Which DNS server do you use?

AdGuard DNS

### DNS protocol

DNS-over-HTTPS

### Environment

```markdown
- OS: Android 14
- Device: OnePlus Nord 4
```

### Root access

- [ ] yes, I have it

### Issue Details

Steps to reproduce:
1. Install and enable AdGuard Android App and Set to Integrated Mode
2. Install AdGuard VPN and connect to any server
3. Enable "Always On VPN" and "Block All Connections Not Using VPN" in Android Settings

### Expected Behavior

Enables Kill Switch functionality, blocking access to the internet when VPN connection drops

### Actual Behavior

AdGuard VPN is unable to connect with an error saying "Network is unavailable. Please connect to the network"

### Screenshots

<details><summary>Screenshot 1</summary>

</details>

### Additional Information

Originally I tried  following the most logical pattern as described "Issue Details" section, meaning I tried enabling AdGuard app, AdGuard VPN, turning on intergrated mode and then Kill Switch via Android settings, as in-app Kill Switch didnt seem to work because of the "You need to enable AdGuard VPN at least once for Kill Switch to start working" prompt

Then I went on to research the issues, the stumbled across this [issue](https://github.com/AdguardTeam/AdGuardVPNForAndroid/issues/110) 

And later went on to additionally try solution described in [this comment](https://github.com/AdguardTeam/AdGuardVPNForAndroid/issues/110#issuecomment-1626145304) in the same thread 

Neither of which worked; AdGuard VPN is still unable to connect to the server unless kill switch is disabled via android settings. Furthermore, AdGuard app now near instantly crashes upon launch, unless I disable protection all together. 

Particularly this part of the original instructions never seemed to work, as I would always get an error "Unable to connect to proxy over TCP"

In the AdGuard app: use AdGuard VPN's SOCKS proxy

Go to settings -> network -> proxy.
Add a new Proxy there, choose "SOCKS5", set proxy host to 127.0.0.1 and proxy port to 1080.
Save it and enable the "Proxy" module.

Can you provide instructions how to set up AdGuard + AdGuard VPN to properly work together with Kill Switch enabled?

**Comment by ESurina** [comment]
In order to troubleshoot this issue, we need to get your application logs.

Here's what we need you to do:

1. Enable debug logging:
*Settings -> General -> Advanced -> Logging level -> Extended*.
2. Reproduce the problem, then remember the exact time it happened.
3. Collect logs:
*Settings -> General -> Advanced -> Export logs and system info*.
4. Send this file to [qa@adguard.com](mailto:qa@adguard.com):
    - include `[android]` keyword and `ISSUE_NUMBER` in the subject of your email
    - specify the exact time when the issue occurred

**Comment by znaf37g** [comment]
> In order to troubleshoot this issue, we need to get your application logs.
> 
> Here's what we need you to do:
> 
> 1. Enable debug logging:
>    _Settings -> General -> Advanced -> Logging level -> Extended_.
> 2. Reproduce the problem, then remember the exact time it happened.
> 3. Collect logs:
>    _Settings -> General -> Advanced -> Export logs and system info_.
> 4. Send this file to [qa@adguard.com](mailto:qa@adguard.com):
>    
>    * include `[android]` keyword and `ISSUE_NUMBER` in the subject of your email
>    * specify the exact time when the issue occurred

I don't think you're getting what I'm asking, so let me rephrase: How do I enable kill switch in Adguard VPN in integrated mode working alongside Adguard app?

When I enable kill switch for Adguard app, VPN stops working.

When I enable kill switch for Adguard VPN, Adguard app stops working.

**Comment by joedownloads1** [comment]
Hi

I am having exact same issue. I updated to adguard nightly to see if it would work but it doesnt. 

I am sending you Logging level -> Extended now from vpn and adguard apk. 
I used a screenshot to mark the time when the problem starts: 10:14 am

![Screenshot_20241229_101206_Settings](https://github.com/user-attachments/assets/33ee66c8-cc30-4dfd-b926-193103c60ce6)


[adguard_vpn_logs_2.10.145_2912_101419.zip](https://github.com/user-attachments/files/18268979/adguard_vpn_logs_2.10.145_2912_101419.zip)

[adguard_logs_4.8.6_2912_101431.zip](https://github.com/user-attachments/files/18268978/adguard_logs_4.8.6_2912_101431.zip)

**Comment by ESurina** [comment]
Ideally, the instructions from the issue #110  should work, but I see that now it does not work and we are figuring out why and what can be done. 
This part of problem: _Particularly this part of the original instructions never seemed to work, as I would always get an error "Unable to connect to proxy over TCP"_ does not reproduce for me, so we would like to get your logs with reproduced error.

**Comment by ESurina** [comment]
Duplicate of #110
